test/events_test.exs

```diff
@@ -1,3 +1,120 @@
 defmodule EventsTest do
   use ExUnit.Case
+
+  defmodule TestSubscriber do
+    def echo_event(event, value, %{send_to: pid} = config) do
+      send(pid, {:event, event, value, config})
+    end
+
+    def raise_on_event(_event, _value, _config) do
+      raise "Got an event"
+    end
+  end
+
+  setup do
+    subscription_id = :crypto.strong_rand_bytes(16) |> Base.encode16()
+
+    on_exit fn ->
+      Events.unsubscribe(subscription_id)
+    end
+
+    {:ok, sub_id: subscription_id}
+  end
+
+  test "subscribing returns error if subscription with the same ID already exist", %{
+    sub_id: sub_id
+  } do
+    :ok = Events.subscribe(sub_id, [:some, :event], TestSubscriber, :echo_event)
+
+    assert :error = Events.subscribe(sub_id, [:some, :event], TestSubscriber, :echo_event)
+  end
+
+  test "subscribed mf is called when event exactly matches the subscription prefix", %{
+    sub_id: sub_id
+  } do
+    event = [:a, :test, :event]
+    config = %{send_to: self()}
+    value = 1
+    Events.subscribe(sub_id, event, TestSubscriber, :echo_event, config)
+
+    Events.emit(event, value)
+
+    assert_receive {:event, ^event, ^value, ^config}
+  end
+
+  test "subscribed mf is not called when event is a prefix of the subscription prefix", %{
+    sub_id: sub_id
+  } do
+    event = [:a, :test]
+    prefix = event ++ [:prefix]
+    config = %{send_to: self()}
+    value = 1
+    Events.subscribe(sub_id, prefix, TestSubscriber, :echo_event, config)
+
+    Events.emit(event, value)
+
+    assert {:messages, []} == Process.info(self(), :messages)
```

Comment:
Is there any advantage of checking it like this vs. `refute_receive`?

Reply:
When asserting with `==` isn't it more readable to have the function call on the left hand side, and the expected result on the right hand side?

Reply:
I like to have expected value on the left hand side of assertion, no matter what assertion I'm using. Do you think it makes it unreadable?

Reply:
I recall some sort of convention, so that's it's more clear (without checking if it's either `=` or `==`) if we're asserting a match (left hand side) or exact equality (right hand side).

It's mentioned here:
https://github.com/christopheradams/elixir_style_guide#testing
But I'm not sure if it's popular.